core/src/test/java/org/elasticsearch/search/highlight/HighlightBuilderTests.java

```diff
@@ -287,44 +292,73 @@ public MappedFieldType fieldMapper(String name) {
         for (int runs = 0; runs < NUMBER_OF_TESTBUILDERS; runs++) {
             HighlightBuilder highlightBuilder = randomHighlighterBuilder();
             SearchContextHighlight highlight = highlightBuilder.build(mockShardContext);
-            XContentBuilder builder = XContentFactory.contentBuilder(randomFrom(XContentType.values()));
-            if (randomBoolean()) {
-                builder.prettyPrint();
-            }
-            builder.startObject();
-            highlightBuilder.innerXContent(builder);
-            builder.endObject();
-            XContentParser parser = XContentHelper.createParser(builder.bytes());
-
-            SearchContextHighlight parsedHighlight = new HighlighterParseElement().parse(parser, mockShardContext);
-            assertNotSame(highlight, parsedHighlight);
-            assertEquals(highlight.globalForceSource(), parsedHighlight.globalForceSource());
-            assertEquals(highlight.fields().size(), parsedHighlight.fields().size());
-
-            Iterator<org.elasticsearch.search.highlight.SearchContextHighlight.Field> iterator = parsedHighlight.fields().iterator();
             for (org.elasticsearch.search.highlight.SearchContextHighlight.Field field : highlight.fields()) {
-                org.elasticsearch.search.highlight.SearchContextHighlight.Field otherField = iterator.next();
-                assertEquals(field.field(), otherField.field());
+                String encoder = highlightBuilder.encoder() != null ? highlightBuilder.encoder() : HighlightBuilder.DEFAULT_ENCODER;
+                assertEquals(encoder, field.fieldOptions().encoder());
+                HighlightBuilder.Field tmp = null;
+                for (HighlightBuilder.Field hbfield : highlightBuilder.fields()) {
+                    if (hbfield.name().equals(field.field())) {
+                        tmp = hbfield;
+                    }
+                }
+                final HighlightBuilder.Field fieldBuilder = tmp;
```

Comment:
Probably worth pulling the "get by name" behavior into a static function in the test. It'll remove the need for `tmp` because you can use early returns and it'll be easier to read this method.